redis/src/caching/statistics.rs

```diff
@@ -0,0 +1,53 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+/// CacheStatistics holds statistics generated by Client Side Caching.
+#[derive(Debug, Default)]
+pub struct CacheStatistics {
```

Comment:
For sure it makes easier to test things, but my main goal for this to be used by users. As an user if I have enabled client side caching in my application, I have no idea how well it works rather than seeing latency decrease etc. But by using CacheStatistics I can generate metrics from it.